ALVRClient/Renderer.swift

```diff
@@ -863,79 +660,30 @@ class Renderer {
         }
     }
     
-    // TODO: figure out how stable Apple's predictions are into the future
-    static let maxPrediction = 30 * NSEC_PER_MSEC
-    static let deviceIdHead = alvr_path_string_to_id("/user/head")
-    
-    func sendTracking(targetTimestamp: Double) {
-        //let targetTimestamp = CACurrentMediaTime() + Double(min(alvr_get_head_prediction_offset_ns(), Renderer.maxPrediction)) / Double(NSEC_PER_SEC)
-        var targetTimestampWalkedBack = targetTimestamp
-        var deviceAnchor:DeviceAnchor? = nil
-        
-        // Predict as far into the future as Apple will allow us.
-        for i in 0...20 {
-            deviceAnchor = worldTracking.queryDeviceAnchor(atTimestamp: targetTimestampWalkedBack)
-            if deviceAnchor != nil {
-                break
-            }
-            targetTimestampWalkedBack -= (5/1000.0)
-        }
-        
-        // Fallback.
-        if deviceAnchor == nil {
-            targetTimestampWalkedBack = CACurrentMediaTime()
-            deviceAnchor = worldTracking.queryDeviceAnchor(atTimestamp: targetTimestamp)
-        }
 
-        // Well, I'm out of ideas.
-        guard let deviceAnchor = deviceAnchor else {
-            return
-        }
-        
-        let targetTimestampNS = UInt64(targetTimestampWalkedBack * Double(NSEC_PER_SEC))
-        
-        deviceAnchorsQueue.append(targetTimestampNS)
-        if deviceAnchorsQueue.count > 1000 {
-            let val = deviceAnchorsQueue.removeFirst()
-            deviceAnchorsDictionary.removeValue(forKey: val)
-        }
-        deviceAnchorsDictionary[targetTimestampNS] = deviceAnchor.originFromAnchorTransform
-        let orientation = simd_quaternion(deviceAnchor.originFromAnchorTransform)
-        let position = deviceAnchor.originFromAnchorTransform.columns.3
-        var trackingMotion = AlvrDeviceMotion(device_id: Renderer.deviceIdHead, orientation: AlvrQuat(x: orientation.vector.x, y: orientation.vector.y, z: orientation.vector.z, w: orientation.vector.w), position: (position.x, position.y, position.z), linear_velocity: (0, 0, 0), angular_velocity: (0, 0, 0))
-        let targetTimestampReqestedNS = UInt64(targetTimestamp * Double(NSEC_PER_SEC))
-        let currentTimeNs = UInt64(CACurrentMediaTime() * Double(NSEC_PER_SEC))
-        //print("asking for:", targetTimestampNS, "diff:", targetTimestampReqestedNS&-targetTimestampNS, "diff2:", targetTimestampNS&-lastRequestedTimestamp, "diff3:", targetTimestampNS&-currentTimeNs)
-        lastRequestedTimestamp = targetTimestampNS
-        alvr_send_tracking(targetTimestampNS, &trackingMotion, 1)
+    // Generic matrix math utility functions
+    func matrix4x4_rotation(radians: Float, axis: SIMD3<Float>) -> matrix_float4x4 {
```

Comment:
These don't need to be instance members of Renderer. They should either be top-level functions like the Xcode template, or removed altogether.